tests/integration_tests/platform.py

```diff
@@ -0,0 +1,66 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+from collections import namedtuple
+
+from tests.integration_tests import integration_settings
+from tests.integration_tests.clouds import (
+    IntegrationCloud,
+    Ec2Cloud,
+    GceCloud,
+    AzureCloud,
+    OciCloud,
+    LxdContainerCloud,
+)
+
+from tests.integration_tests.instances import (
+    CloudInstance,
+    Ec2Instance,
+    GceInstance,
+    AzureInstance,
+    OciInstance,
+    LxdContainerInstance,
+)
+
+platform = namedtuple('Platform', 'cloud instance')
+platforms = {
+    'ec2': platform(Ec2Cloud, Ec2Instance),
+    'gce': platform(GceCloud, GceInstance),
+    'azure': platform(AzureCloud, AzureInstance),
+    'oci': platform(OciCloud, OciInstance),
+    'lxd_container': platform(LxdContainerCloud, LxdContainerInstance),
+}
+
+if integration_settings.PLATFORM not in platforms.keys():
+    raise ValueError(
+        "{} is an invalid PLATFORM specified in settings. "
+        "Must be one of {}".format(
+            integration_settings.PLATFORM, list(platforms.keys())
+        )
+    )
+
+
+_session_cloud = None
+_dynamic_instance = None
+
+
+def initialize_session_client():
+    global _session_cloud
+    global _dynamic_instance
```

Comment:
I'm happy for us to retain these and their associated check, but I would lean towards trusting pytest's session-scoping to only call this once for us. (Just noting this in case you were also considering that and this nudges your opinion in the other direction. :+1:)

Reply:
I'm not sure if this is still an issue, but can you describe what you mean by truesting pytest's session-scoping? For the `_session_cloud` variable, we need to access it in more than once place (and potentially in tests themselves), so I think the module-level variable with the function best achieves that.